implementations/rust/ockam/ockam_app/src/invitations/tray_menu.rs

```diff
@@ -0,0 +1,138 @@
+use tauri::{
+    AppHandle, CustomMenuItem, SystemTrayMenu, SystemTrayMenuItem, SystemTraySubmenu, Wry,
+};
+use tracing::debug;
+
+use ockam_api::cloud::share::{ReceivedInvitation, SentInvitation};
+
+use crate::{app::AppState, invitations::InvitationState};
+
+pub(crate) const INVITATIONS_SENT_HEADER_MENU_ID: &str = "sent_invitations_header";
+pub(crate) const INVITATIONS_RECEIVED_HEADER_MENU_ID: &str = "received_invitations_header";
+pub(crate) const INVITATIONS_MANAGE_MENU_ID: &str = "manage_invites";
+pub(crate) const INVITATIONS_WINDOW_ID: &str = "share";
+
+pub(crate) async fn build_invitations_section(
+    app_state: &AppState,
+    tray_menu: SystemTrayMenu,
+) -> SystemTrayMenu {
+    if !app_state.is_enrolled().await {
+        return tray_menu;
+    };
+
+    let InvitationState { sent, received } = app_state.model(|f| f.get_invitations()).await;
+    debug!(?sent, ?received);
+
+    let mut tray_menu = tray_menu.add_native_item(SystemTrayMenuItem::Separator);
+    tray_menu = add_sent_menu(tray_menu, sent);
+    tray_menu = add_received_menu(tray_menu, received);
+    // TODO: Add active shares with connect/leave verbs
+    tray_menu.add_item(CustomMenuItem::new(
+        INVITATIONS_MANAGE_MENU_ID,
+        "Manage Invitations...",
+    ))
+}
+
+fn add_sent_menu(tray_menu: SystemTrayMenu, sent: Vec<SentInvitation>) -> SystemTrayMenu {
+    let header_text = if sent.is_empty() {
+        "No Sent Invitations"
+    } else {
+        "Sent Invitations"
+    };
+    sent.iter().map(sent_invitation_menu).fold(
+        tray_menu
+            .add_item(CustomMenuItem::new(INVITATIONS_SENT_HEADER_MENU_ID, header_text).disabled()),
+        |menu, submenu| menu.add_submenu(submenu),
+    )
+}
+
+fn sent_invitation_menu(invitation: &SentInvitation) -> SystemTraySubmenu {
+    let id = invitation.id.to_owned();
+    // TODO: List creation date
+    // TODO: List target scope and ID
+    let submenu = SystemTrayMenu::new()
+        .add_item(CustomMenuItem::new(id.clone(), id.clone()).disabled())
+        .add_item(
+            CustomMenuItem::new(
+                id.clone(),
+                invitation
+                    .recipient_email
+                    .as_ref()
+                    .map(|e| format!("Recipient email: {e}"))
+                    .unwrap_or_else(|| "Recipient email: Unset".to_string()),
+            )
+            .disabled(),
+        )
+        .add_item(
+            CustomMenuItem::new(
+                id.clone(),
+                format!("Remaining uses: {}", invitation.remaining_uses),
+            )
+            .disabled(),
+        )
+        .add_item(CustomMenuItem::new(
+            format!("cancel-sent-{}", invitation.id),
+            "Cancel",
```

Comment:
What is this exactly? There is no event handler attached to it right? At first sight I thought it was a button to exit the submenu, but I'm not sure now.

Reply:
Correct, there's no event handler attached yet. It will be used by the owner to revoke an outstanding invitation so that it can no longer be used and will no longer be listed for the recipient's UI.